### ISSUE TITLE
Failed to launch/connect server at startup after insightface workaround on Windows

### ISSUE THREAD

**Issue by ccoroner** [Acly/krita-ai-diffusion#1199]
Hello,
this plugin looks awesome! Sadly, I cannot get it to work and there is an error when starting up.

Setup:
- I downloaded the plugin and for use of insightface, I edited the server.py to set `whl_file = self._cache_dir / "insightface-0.7.3-cp311-cp311-win_amd64.whl"` and `whl_url = "https://github.com/Gourieff/Assets/raw/main/Insightface/insightface-0.7.3-cp311-cp311-win_amd64.whl"` as I found browsing the issues here and got the "cannot build wheel" error, before.
- I installed the plugin. Everything went fine!

When starting Krita, the client.log catches the following error:
`2024-09-19 10:33:19,102 INFO Loaded 11 sampler presets from userpath\AppData\Roaming\krita\pykrita\ai_diffusion\presets\samplers.json
2024-09-19 10:33:19,141 INFO Extension initialized, Version: 1.24.0, Python: 3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)]
2024-09-19 10:33:19,142 INFO Loading settings from userpath\AppData\Roaming\krita\ai_diffusion\settings.json
2024-09-19 10:33:19,144 INFO Found server installation v1.24.0 at userpath\AppData\Roaming\krita\ai_diffusion\server
2024-09-19 10:33:19,155 INFO Loaded 2 model files from userpath\AppData\Roaming\krita\ai_diffusion\database\loras.json
2024-09-19 10:33:23,110 INFO Starting server with python -su -Xutf8 main.py
2024-09-19 10:33:23,405 ERROR Error during server start: [WinError 87] Falscher Parameter
Traceback (most recent call last):
  File "userpath\AppData\Roaming\krita\pykrita\ai_diffusion\server.py", line 390, in start
    self._process = await create_process(
  File "userpath\AppData\Roaming\krita\pykrita\ai_diffusion\util.py", line 211, in create_process
    p = await asyncio.create_subprocess_exec(
  File "asyncio\subprocess.py", line 218, in create_subprocess_exec
  File "asyncio\base_events.py", line 1667, in subprocess_exec
  File "asyncio\windows_events.py", line 394, in _make_subprocess_transport
  File "asyncio\base_subprocess.py", line 36, in __init__
  File "asyncio\windows_events.py", line 890, in _start
  File "asyncio\windows_utils.py", line 153, in __init__
  File "subprocess.py", line 969, in __init__
  File "subprocess.py", line 1438, in _execute_child
OSError: [WinError 87] Falscher Parameter
2024-09-19 10:33:23,408 WARNING Failed to launch/connect server at startup: [WinError 87] Falscher Parameter
`

Additionally, I logged the parameters that are called in `server.py`:
`2024-09-19 17:57:22,017 INFO _python_cmd: userpath\AppData\Roaming\krita\ai_diffusion\server\python\python.exe
2024-09-19 17:57:22,020 INFO args: ['-su', '-Xutf8', 'main.py']
2024-09-19 17:57:22,020 INFO comfy_dir: userpath\AppData\Roaming\krita\ai_diffusion\server\ComfyUI
2024-09-19 17:57:22,020 INFO env: {'ONEDNN_MAX_CPU_ISA': 'AVX2'}`
That seems to be okay.

I noticed that the error ` [WinError 87] Falscher Parameter` (meaning ` [WinError 87] parameter incorrect` happens when a process or subprocess is called incorrectly. However, the log also mentions Python: 3.10.7 while I got Python 3.11.9 installed.
I guess that this version mismatch might cause the issue but I could not find where the plugin refers to python 3.10. anywhere else than in the server.py

Could you give me a hint on how to resolve this issue, please?
Thanks in advance!

**Comment by paginaweb123** [comment]
maybe this will help
https://bit.ly/4evXrQ6

you may need to install the c compiler

**Comment by Acly** [comment]
I don't think that error has anything to do with python version or insightface either. It's an issue with act of launching a process itself, not so much the particular process it's trying to launch.

Not really sure about the cause though, it might be related to Windows version, security settings, or a Anti-virus intercepting the launch (they tend to cause weird errors like this).

As to the "No checkpoints found!" - just to be sure, you did install one of the models in the checkpoint seciont...?

**Comment by ccoroner** [comment]
Thank you for your reply!

You are absolutely right! I got the "Custom Server (local or remote)" option to work just a few minutes ago after installing everything manually, including the models and checkpoints. So that is finally up and running. :)

I probably messed up a part of the installation before. By doing the installation manually I learned that the `[WinError 87]` I got was due to a missing directory in ComfyUI and maybe odd git settings.

I will try to remove everything and do a clean install for the "Local Managed Server" again.
So, my apologies for opening an issue here! This is not an issue of the plugin at all. Just frustrating Windows stuff.

Really awesome work and plugin!
